### ISSUE TITLE
Ambiguous error when booleans are passed to hoplon.

### ISSUE THREAD

**Issue by flyboarder** [hoplon/hoplon#228]
Booleans should convert themselves to strings and implement INode? Also need to enhance specs to catch this type of runtime issue.

**Comment by thedavidmeister** [comment]
@flyboarder could you post an example of what you're expecting to see here?

**Comment by flyboarder** [comment]
Ideally our protocols should implement all of the data types so that there is a standard way of handling them.

**Comment by thedavidmeister** [comment]
yeah, that error is cryptic but i'm not sure what a boolean should do, other than throw a clearer error?

**Comment by flyboarder** [comment]
I would suggest a good error and empty string. Also better catch in the spec, though I don’t have runtime specs turned on so I’m not sure what it does right now.

**Comment by flyboarder** [comment]
So quick look over the spec, it assumes all data types are valid children, we could modify this and make it something that could be caught by runtime instrumentation.

**Comment by thedavidmeister** [comment]
@flyboarder yeah that would be great, i don't see any reason to try and force incompatible data types to work but some more helpful error messages would be a big win
